controllers/glance_controller.go

```diff
@@ -594,92 +594,128 @@ func (r *GlanceReconciler) reconcileNormal(ctx context.Context, instance *glance
-
-	// Get internal GlanceAPI's condition status for comparison with external below
-	internalAPICondition := glanceAPI.Status.Conditions.Mirror(glancev1.GlanceAPIReadyCondition)
+	instance.Status.Conditions.MarkTrue(condition.CronJobReadyCondition, condition.CronJobReadyMessage)
+	// create CronJob - end
+	return ctrl.Result{}, nil
+}
 
-	// Regardless of what the user may have set in GlanceAPIExternal.EndpointType,
-	// we force "external" here by passing glancev1.APIExternal for the apiType arg
-	glanceAPI, op, err = r.apiDeploymentCreateOrUpdate(ctx, instance, instance.Spec.GlanceAPIExternal, glancev1.APIExternal, helper)
+// apiDeployment represents the logic of deploying GlanceAPI instances specified
+// in the main CR according to a given strategy (split vs single). It handles
+// the deployment logic itself, as well as the output settings mirrored in the
+// main Glance CR status
+func (r *GlanceReconciler) apiDeployment(ctx context.Context, instance *glancev1.Glance, current glancev1.GlanceAPITemplate, helper *helper.Helper) error {
+
+	// By default internal and external points to diff instances, but we might
+	// want to override "external" with "single" in case APIType == "single":
+	// in this case we only deploy the External instance and skip the internal
+	// one
+	var internal string = glancev1.APIInternal
+	var external string = glancev1.APIExternal
+
+	// If we're deploying a "single" instance, we skip GlanceAPI.Internal, and
+	// we only deploy the External instance passing "glancev1.APISingle" to the
+	// GlanceAPI controller, so we can properly handle this use case (nad and
+	// service creation).
+	if current.Type == "single" {
+		external = glancev1.APISingle
+	}
+	glanceAPI, op, err := r.apiDeploymentCreateOrUpdate(ctx, instance, instance.Spec.GlanceAPI, external, helper)
 	if err != nil {
 		instance.Status.Conditions.Set(condition.FalseCondition(
 			glancev1.GlanceAPIReadyCondition,
 			condition.ErrorReason,
 			condition.SeverityWarning,
 			glancev1.GlanceAPIReadyErrorMessage,
 			err.Error()))
-		return ctrl.Result{}, err
+		return err
 	}
 	if op != controllerutil.OperationResultNone {
 		r.Log.Info(fmt.Sprintf("Deployment %s successfully reconciled - operation: %s", instance.Name, string(op)))
 	}
 
-	// Mirror external GlanceAPI status' APIEndpoints and ReadyCount to this parent CR
+	// Mirror single/external GlanceAPI status' APIEndpoints and ReadyCount to this parent CR
 	if glanceAPI.Status.APIEndpoints != nil {
 		instance.Status.APIEndpoints[string(endpoint.EndpointPublic)] = glanceAPI.Status.APIEndpoints[string(endpoint.EndpointPublic)]
+		// if we don't split, both apiEndpoints (public and internal) should be
+		// reflected to the main Glance CR
+		if current.Type == "single" {
+			instance.Status.APIEndpoints[string(endpoint.EndpointInternal)] = glanceAPI.Status.APIEndpoints[string(endpoint.EndpointInternal)]
+		}
 	}
 	instance.Status.GlanceAPIExternalReadyCount = glanceAPI.Status.ReadyCount
```

Comment:
Is there any reason we might want to set `instance.Status.GlanceAPIInternalReadyCount` here as well if type is `single`?

Reply:
In theory a `single` instance acts as both `Internal` and `External`, and I though here two things:
1. if type is `single`, setting the `ReadyCount` in the `Status` is something good to have and aligned with the `split` use case (where we set two of them);
2. we can reuse the same variable as we don't really need a different one that should be mirrored in the main CR status, but it's something we can actually do (making this logic a little bit more verbose);

The other option is not setting it at all when instance is `single`, as we only check this info in the `kuttl` assertion, and afaik is not used anywhere to mark the glance condition as Ready.

Do you think that it makes sense to add a `GlanceAPIReadyCount int32 `json:"glanceAPIReadyCount,omitempty"` to [1] ?

[1] https://github.com/openstack-k8s-operators/glance-operator/blob/main/api/v1beta1/glance_types.go#L155

Reply:
I think it's fine reusing the variables as you've done for the meantime.  To your point, they are mostly just informative and aren't considered in any calculation.  This was just another case of my curiosity posing a question. :)